clients/python/dd_image_client.py

```diff
@@ -0,0 +1,316 @@
+import base64
+import time
+import json
+import os
+from dd_client import DD
+import cStringIO
+from PIL import Image
+from PIL import ExifTags
+
+
+from scipy import spatial
+
+
+class DDImage(object):
+
+    mllib = 'caffe'
+
+    def __init__(self, host):
+
+        self.host = host
+        self.sname = ''
+        self.description = ''
+        self.mltype = ''
+        self.extract_layer = ''
+        self.nclasses = 0
+        self.width = self.height = 224
+        self.binarized = False
+        self.GPU = True
+        self.model_repo = ''
+        self.dd = None
+        self.model_name = ''
+        self.batch_size = None
+        self.GOOGLENET_BATCH_SIZE = 40
+        self.RESNET152_BATCH_SIZE = 4
+        self.model_repo
+
+    def init_for_classification(self, model_name, model_repo):
+        self.sname = "_classification"
+        self.description = self.sname + ' for classification'
+        self.mltype = 'supervised'
+        self.nclasses = 1000
+        self.model_name = model_name
+        self.model_repo=model_repo
+        self.dd = DD(self.host)
+        self.dd.set_return_format(self.dd.RETURN_PYTHON)
+
+    def init_for_extraction(self, model_name, model_repo):
+        self.sname = model_name + "_extraction"
+        self.description = self.sname + ' for feature extraction'
+        self.mltype = 'unsupervised'
+        self.model_name = model_name
+        self.model_repo = model_repo
+
+        if model_name == 'resnet152':
+            self.extract_layer = 'fc1000'
+            self.nclasses = 1000
+            self.batch_size = self.RESNET152_BATCH_SIZE
+        elif model_name == 'googlenet':
+            self.extract_layer = 'pool5/7x7_s1'
+            self.nclasses = 1000
+            self.batch_size = self.GOOGLENET_BATCH_SIZE
+        else:
```

Comment:
The implementation is tied to the models that DD currently comes with. I agree it may be a good idea to add a more generalized one, but that wasn't the intent of this implementation. I wanted people to be able to start using the client with 3 lines of code which is why the init methods basically take care of everything and fail otherwise.